emmaa/model.py

```diff
@@ -945,6 +947,119 @@ def get_search_term_names(model, bucket=EMMAA_BUCKET_NAME):
+    stmts : list[indra.statements.Statement]
+        A list of statements to load evidence for.
+    method : str
+        What method to use to load evidence (accepted values: db_query and
+        rest_api). Default: db_query.
+    ev_limit : Optional[int]
+        How many evidences to load from the database for each statement.
+        Default: 1000.
+    batch_size : Optional[int]
+        Batch size used for querying. Default: 3000.
+
+    Returns
+    -------
+    stmts : list[indra.statements.Statement]
+        A list of statements with additional evidence.
+    """
+    stmt_hashes = [stmt.get_hash() for stmt in stmts]
+    # get stmts from database
+    logger.info(f'Loading additional evidences for {len(stmts)} stmts from db'
+                f' using {method} method')
+    new_stmts = []
+    offset = 0
+    while True:
+        if batch_size:
+            hashes_batch = stmt_hashes[offset:(offset + batch_size)]
+        else:
+            hashes_batch = stmt_hashes
+        if method == 'rest_api':
+            proc = get_statements_by_hash(hashes_batch, ev_limit=ev_limit)
+            batch_stmts = proc.statements
+        elif method == 'db_query':
+            q = HasHash(hashes_batch)
+            res = q.get_statements(ev_limit=ev_limit)
+            batch_stmts = res.statements()
+        new_stmts += batch_stmts
+        offset += batch_size
+        if offset >= len(stmt_hashes):
+            break
+    logger.info(f'Found {len(new_stmts)} stmts in the database')
+    evid_by_hash = {stmt.get_hash(): stmt.evidence for stmt in new_stmts}
+    # add db evidence to emmaa stmts evidence
+    # this can create duplicates but they are handled by preassembly
+    for stmt in stmts:
+        stmt.evidence += evid_by_hash.get(stmt.get_hash(), [])
+    return stmts
+
+
+@register_pipeline
+def run_preassembly_with_extra_evidence(stmts_in, return_toplevel=True,
+                                        belief_scorer=None,
+                                        query_method='db_query', ev_limit=10,
+                                        batch_size=3000, **kwargs):
+    """Run preassembly on a list of statements.
+
+    Parameters
+    ----------
+    stmts_in : list[indra.statements.Statement]
+        A list of statements to preassemble.
+    return_toplevel : Optional[bool]
+        If True, only the top-level statements are returned. If False,
+        all statements are returned irrespective of level of specificity.
+        Default: True
+    belief_scorer : Optional[indra.belief.BeliefScorer]
+        Instance of BeliefScorer class to use in calculating Statement
+        probabilities. If None is provided (default), then the default
+        scorer is used.
+    query_method : str
+        What method to use to load evidence (accepted values: db_query and
+        rest_api). Default: db_query.
+    ev_limit : Optional[int]
+        How many evidences to load from the database for each statement.
+        Default: 1000.
```

Comment:
The default in the docstring doesn't match the ev_limit in the function definition

Reply:
Updated it to 1000 everywhere